src/hole.rs

```diff
@@ -95,8 +95,10 @@ impl HoleList {
     /// This function walks the list and inserts the given block at the correct place. If the freed
     /// block is adjacent to another free block, the blocks are merged again.
     /// This operation is in `O(n)` since the list needs to be sorted by address.
```

Comment:
We need to update the docs here too, e.g. with: "The function performs exactly the same layout adjustments as [`allocate_first_fit`] and returns the aligned layout."